### PR TITLE
nsm without next

### PR DESCRIPTION
**Pull request by itelo**
NSM is currently utilizing only the `pages-manifest.json` file from Next.js. This file is essentially a JSON object that contains keys representing the endpoints and their corresponding file path values.

`page-manifest.json`

```json
{
  "/api/public": "pages/api/public/index.ts",
  "/api/public/workspaces/reset_sandbox": "pages/api/public/workspaces/reset_sandbox.ts",
}
```

The primary focus of this PR is to develop a method for generating the `pages-manifest.json` without relying on nextjs and integrating it with `ts-up`. It's crucial to note that this approach is only suitable for API endpoints, and if the project involves react or static files, it may lead to errors. To prevent this, I've included an alert.

### CODE REVIEW

File: lib/cli.js
Comment (seveibar) [inline lib/cli.js:36]: Log lines

File: tests/fixtures/get-sample-project-without-next.ts
Comment (seveibar) [inline tests/fixtures/get-sample-project-without-next.ts:17]: getNSMIndex

**Review by seveibar — APPROVED** [review]
Generally looks good but we should address naming cc @razor-x @codetheweb this allows you to build nsm modules without next

**Comment by seveibar** [comment]
I would also rename the mode for building without next to "nextless", that could be the flag passed to nsm

**Review by seveibar — COMMENTED** [review]
Small bug: skip-next-build should become "skip-build" and "nextless" should be the build without next

File: tests/fixtures/get-sample-project-without-next.ts
Comment (razor-x) [reply tests/fixtures/get-sample-project-without-next.ts:17]: `getNsmIndex`?

File: package.json
Comment (razor-x) [inline package.json:74]: Where is this used?

File: nsm/scripts/generate-routes.js
Comment (razor-x) [inline nsm/scripts/generate-routes.js:12]:
```suggestion
  const pageFiles = glob.sync("**/*.{ts,tsx,js,jsx,cjs,mjs,mjsx,cjsx}", { cwd: pagesDir })
```

File: nsm/scripts/generate-routes.js
Comment (razor-x) [inline nsm/scripts/generate-routes.js:22]: Can we put the valid extensions in an array const at the top and reuse them via `.join(',')` in the other spots they show up?

File: nsm/scripts/generate-routes.js
Comment (razor-x) [inline nsm/scripts/generate-routes.js:112]: We should respect the consumers prettier config if possible.

**Review by razor-x — CHANGES_REQUESTED** [review]
Can you explain how the new build works? That will help me properly evaluate this and others understand what we are doing here. Otherwise I agree about the naming and other cleanup.

**Comment by itelo** [comment]
> Can you explain how the new build works? That will help me properly evaluate this and others understand what we are doing here. Otherwise I agree about the naming and other cleanup.
> 
> It's pretty neat you were able to figure this out!

thanks for the feedback, I have update the PR description :)

File: package.json
Comment (itelo) [reply package.json:74]: ts-up is complaining about it

Comment (itelo) [reply package.json:74]: ts-up was added to bundle when we decide to build without nextjs

File: nsm/scripts/generate-routes.js
Comment (itelo) [reply nsm/scripts/generate-routes.js:114]: I personally don't think it's a big deal since the file will be saved in .nsm and the consumers won't be looking at it.

File: lib/build.js
Comment (razor-x) [inline lib/build.js:8]: This is a breaking change right? `skipNextBuild => skipBuild`. We will need to keep the old option and deprecate it first.

File: lib/build.js
Comment (razor-x) [inline lib/build.js:10]: Should we add defaults for all of these bools?
```suggestion
  skipBuild = false,
  nextless= false,
  onlyApiFiles = false,
```

File: lib/build.js
Comment (razor-x) [inline lib/build.js:48]: Since `require` is i/o that might fail, we should conditionally execute this. We can move it to a function with early return or use if/else.

File: nsm/scripts/generate-routes.js
Comment (razor-x) [inline nsm/scripts/generate-routes.js:74]: Can this be moved to a pure function? I cannot understand what this is supposed to do.

File: nsm/scripts/generate-routes.js
Comment (razor-x) [inline nsm/scripts/generate-routes.js:102]: Here and in other places, I think we can do what we need with methods from 'node:path' instead of direct string manipulation. It should be safer and more clear.

**Review by razor-x — CHANGES_REQUESTED** [review]
We should resolve the open question on the `skipNextBuild => skipBuild` rename first. Other comments are more code style suggestions.

File: lib/build.js
Comment (itelo) [reply lib/build.js:48]: what if we have something like:
```suggestion
    const getNextjsOptions = () =>  ({
      build: require(`${packageDirWithNext}/node_modules/next/dist/build/index`).default,
      generateRoutes: require(`${dir}/.nsm/scripts/generate-routes.js`).generateRoutes,
    })

    const getNextlessOptions = () => ({
      build: require(`${dir}/.nsm/scripts/nsm-build.js`).nsmBuild,
      generateRoutes: require(`${dir}/.nsm/scripts/generate-routes.js`).generateNsmRoutes, 
    })
```

File: lib/build.js
Comment (itelo) [reply lib/build.js]: I added it as deprecated 
image

File: nsm/scripts/generate-routes.js
Comment (itelo) [reply nsm/scripts/generate-routes.js:111]: Let's create another issue to solve this to not block this PR

File: lib/build.js
Comment (razor-x) [reply lib/build.js]: This option needs to be retained and deprecated here as well, or it's still a breaking change: https://github.com/seamapi/nextjs-server-modules/pull/47/files#diff-1ae50dc56d6f821fbc0183fc424426e7cf420c9f5e0a0a5fd0e1801a573b8fd0R18

**Review by razor-x — APPROVED** [review]
Let's give it a try. Please add the `skipNextBuild` option back before merge.

**Review by seveibar — APPROVED** [review]

**Pull request merged**
